tests/test_client.py

```diff
@@ -55,107 +55,113 @@ def test_transport_option(monkeypatch):
-
-def test_proxy_none_httpenv_select(monkeypatch):
-    monkeypatch.setenv("HTTP_PROXY", "http://localhost/123")
-    client = Client("http://foo@sentry.io/123")
-    assert client.transport._pool.proxy.scheme == "http"
-
-
-def test_proxy_none_httpsenv_select(monkeypatch):
-    monkeypatch.setenv("HTTPS_PROXY", "https://localhost/123")
-    client = Client("https://foo@sentry.io/123")
-    assert client.transport._pool.proxy.scheme == "https"
-
-
-def test_proxy_none_httpenv_fallback(monkeypatch):
-    monkeypatch.setenv("HTTP_PROXY", "http://localhost/123")
-    client = Client("https://foo@sentry.io/123")
-    assert client.transport._pool.proxy.scheme == "http"
-
-
-def test_proxy_bothselect_bothen(monkeypatch):
-    monkeypatch.setenv("HTTP_PROXY", "http://localhost/123")
-    monkeypatch.setenv("HTTPS_PROXY", "https://localhost/123")
-    client = Client("https://foo@sentry.io/123", http_proxy="", https_proxy="")
-    assert client.transport._pool.proxy is None
-
-
-def test_proxy_bothavoid_bothenv(monkeypatch):
-    monkeypatch.setenv("HTTP_PROXY", "http://localhost/123")
-    monkeypatch.setenv("HTTPS_PROXY", "https://localhost/123")
-    client = Client("https://foo@sentry.io/123", http_proxy=None, https_proxy=None)
-    assert client.transport._pool.proxy.scheme == "https"
-
-
-def test_proxy_bothselect_httpenv(monkeypatch):
-    monkeypatch.setenv("HTTP_PROXY", "http://localhost/123")
-    client = Client("https://foo@sentry.io/123", http_proxy=None, https_proxy=None)
-    assert client.transport._pool.proxy.scheme == "http"
-
-
-def test_proxy_httpselect_bothenv(monkeypatch):
-    monkeypatch.setenv("HTTP_PROXY", "http://localhost/123")
-    monkeypatch.setenv("HTTPS_PROXY", "https://localhost/123")
-    client = Client("https://foo@sentry.io/123", http_proxy=None, https_proxy="")
-    assert client.transport._pool.proxy.scheme == "http"
-
-
-def test_proxy_httpsselect_bothenv(monkeypatch):
-    monkeypatch.setenv("HTTP_PROXY", "http://localhost/123")
-    monkeypatch.setenv("HTTPS_PROXY", "https://localhost/123")
-    client = Client("https://foo@sentry.io/123", http_proxy="", https_proxy=None)
-    assert client.transport._pool.proxy.scheme == "https"
-
-
-def test_proxy_httpselect_httpsenv(monkeypatch):
-    monkeypatch.setenv("HTTPS_PROXY", "https://localhost/123")
-    client = Client("https://foo@sentry.io/123", http_proxy=None, https_proxy="")
-    assert client.transport._pool.proxy is None
-
-
-def test_proxy_httpsselect_bothenv_http(monkeypatch):
-    monkeypatch.setenv("HTTP_PROXY", "http://localhost/123")
-    monkeypatch.setenv("HTTPS_PROXY", "https://localhost/123")
-    client = Client("http://foo@sentry.io/123", http_proxy=None, https_proxy=None)
-    assert client.transport._pool.proxy.scheme == "http"
+@pytest.mark.parametrize(
+    "dsn,env_http_proxy,env_https_proxy,arg_http_proxy,arg_https_proxy,expected_proxy_scheme",
+    [
+        ("http://foo@sentry.io/123", None, None, "http://localhost/123", None, "http"),
```

Comment:
I wonder if instead of these tuples we should have one parameter called `testcase` and pass dictionaries into the testcase. I think this is hard to read because you need to memorize which tuple index maps to which param.

Reply:
Done :)